dask_cuda/utils.py

```diff
@@ -649,3 +653,105 @@ def get_gpu_uuid_from_index(device_index=0):
+    plugin_vals = dask_worker.plugins.values()
+    ret = {}
+
+    # device and host memory configuration
+    for p in plugin_vals:
+        for v in dir(p):
+            # ignore hidden attrs as well as setup and CPUAffinity
+            if v.startswith("_") or v == "setup" or v == "cores":
+                pass
+            else:
+                ret[v] = getattr(p, v)
+    for mem in [
+        "memory_limit",
+        "memory_pause_fraction",
+        "memory_spill_fraction",
+        "memory_target_fraction",
+    ]:
+        ret[mem] = getattr(dask_worker.memory_manager, mem)
+
+    # jit unspilling set
+    ret["jit-unspill"] = isinstance(dask_worker.data, ProxifyHostFile)
+
+    # get optional device-memory-limit
+    if ret["jit-unspill"]:
+        ret["device-memory-limit"] = dask_worker.data.manager._device_memory_limit
+    else:
+        has_device = hasattr(dask_worker.data, "device_buffer")
+        if has_device:
+            ret["device-memory-limit"] = dask_worker.data.device_buffer.n
+
+    # using ucx ?
+    scheme, loc = parse_address(dask_worker.scheduler.address)
+    ret["protocol"] = scheme
+    if scheme == "ucx":
+        import ucp
+
+        ret["ucx-transports"] = ucp.get_active_transports()
+
+    # comm timeouts
+    ret["distributed.comm.timeouts"] = dask.config.get("distributed.comm.timeouts")
+
+    return ret
+
+
+async def _get_sched_config(client):
+    worker_ttl = await client.run_on_scheduler(
+        lambda dask_scheduler: dask_scheduler.worker_ttl
+    )
+    extensions = list(
+        await client.run_on_scheduler(
+            lambda dask_scheduler: dask_scheduler.extensions.keys()
+        )
+    )
+    ret = {}
+    ret["distributed.scheduler.worker-ttl"] = worker_ttl
+    ret["active-extensions"] = extensions
+
+    return ret
+
+
+async def _get_cluster_configuration(client):
+    worker_config = await client.run(get_worker_config)
+    ret = await _get_sched_config(client)
+
+    # does the cluster have any workers ?
+    if worker_config:
+        w = list(worker_config.values())[0]
+        ret.update(w)
+
+    return ret
+
+
+def get_cluster_configuration(client, table=False):
+    ret = client.sync(
+        _get_cluster_configuration, client=client, asynchronous=client.asynchronous
+    )
+    if not table:
+        return ret
+    else:
+        try:
+            from rich.console import Console
+            from rich.table import Table
+        except ImportError:
+            print(
+                "Please install rich `python -m pip install rich`"
+                / "to print a table of the current Dask Cluster Configuration"
+            )
+            sys.exit(1)
+
+        table = Table(title="Dask Cluster Configuration")
+        table.add_column("Parameter", justify="left", style="red")
+        table.add_column("Value", justify="left", style="green")
+
+        params = sorted(ret.keys())
+        for p in params:
+            table.add_row(p, str(ret[p]))
```

Comment:
```suggestion
        for p in sorted(ret.keys()):
            table.add_row(p, str(ret[p]))
```
Or:
```
for key, value in sorted(ret.items(), key=operator.itemgetter(0)):
    table.add_row(key, str(value))
```
If you like

Reply:
Aside: why do we need to explicit `str` the value, but not the key? With my proposed change to providing nested dictionaries for providing configuration settings of plugins would suggest one could format as nested tables.

Reply:
Rich uses [renderables](https://rich.readthedocs.io/en/latest/protocol.html#console-render) which are either strings or have implemented valid `__rich__` / `__rich_console__` methods:
https://github.com/Textualize/rich/blob/84e628655a2981ee90413ca3f35001ec3954161d/rich/protocol.py#L10-L16